plugins/inputs/t128_tank/reader.go

```diff
@@ -0,0 +1,364 @@
+		fmt.Errorf("unable to parse index after detecting boundary fault: %s", err)
+		return false, StartIndex
+	}
+
+	return true, nextIndex
+}
+
+func (r *Reader) readFromTank(readCtx context.Context, startingIndex index) (err error) {
+	cmdArgs := []string{
+		"/opt/128technology/bin/tank-cli",
+		"-b",
+		fmt.Sprintf("%s:%d", r.tankAddress, r.tankPort),
+		"-t",
+		r.topic,
+		"consume",
+		"-F",
+		"seqnum,content",
+		startingIndex.string(),
+	}
+
+	cmd := r.tankReadCmdCtx(readCtx, cmdArgs[0], cmdArgs[1:]...)
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("unable to pipe %s tank output: %s", r.topic, err)
+	}
+
+	tankReader := bufio.NewReader(stdout)
+	if err := cmd.Start(); err != nil {
+		return fmt.Errorf("unable to start %s tank stream: %w", r.topic, err)
+	}
+
+	var parseErr error
+	var messages []*IndexedMessage
+
+	for {
+		select {
+		case <-readCtx.Done():
+			r.log.Errorf("%s read routine done", r.topic)
+			return nil
+		default:
+		}
+
+		messages, parseErr = parseLines(tankReader, r.topic)
+		if parseErr != nil {
+			r.log.Errorf("encountered error while parsing lines in %s output, will not attempt to parse more lines: %v", r.topic, parseErr)
+			return parseErr
+		}
+		var collectedMessages []IndexedMessage
+		for _, message := range messages {
+			if message.IsValid() {
+				collectedMessages = append(collectedMessages, *message)
+			}
+		}
+		select {
+		case <-readCtx.Done():
+			return nil
+		case r.sendChan <- collectedMessages:
+		}
+	}
+}
+
+func getIndex(indexPath string, defaultIndex index) (index, error) {
+	if indexPath == "" {
+		log.Printf("index file path not provided, starting with default index %s", defaultIndex.string())
+		return defaultIndex, nil
+	}
+	content, err := os.ReadFile(indexPath)
+	if errors.Is(err, os.ErrNotExist) {
+		log.Printf("index file %s does not exist, starting with default index %s", indexPath, defaultIndex.string())
+		return defaultIndex, nil
+
+	} else if err != nil {
+		return defaultIndex, fmt.Errorf("encountered error reading index file, starting with default index %s: %s", defaultIndex.string(), err)
+	}
+
+	log.Printf("found '%s' in index file", content)
+
+	index, err := newIndex(string(content))
+	if err != nil {
+		return defaultIndex, fmt.Errorf("encountered error while parsing index file content, starting with default index %s: %s", defaultIndex.string(), err)
+	}
+
+	return index, nil
+}
+
+func setIndex(indexPath string, index index) {
+	if indexPath == "" {
+		log.Printf("index file path not provided, starting with index %d", index.value)
+		return
+	}
+	_, err := os.Stat(path.Dir(indexPath))
+	if errors.Is(err, os.ErrNotExist) {
+		err := os.MkdirAll(path.Dir(indexPath), os.ModePerm)
+		if err != nil {
+			log.Printf("unable to create index file parent directory: %s", err)
+			return
+		}
+	}
+
+	err = os.WriteFile(indexPath, []byte(index.string()), 0644)
+	if err != nil {
+		log.Printf("unable to update index to %d: %s", index, err)
```

Comment:
A bunch of logs are using the default logger instead of the telegraf logger. They all need to use the telegraf logger.